File: python-algo/leetcode/problem_1894.py
"""1894. Find the Student that Will Replace the Chalk
https://leetcode.com/problems/find-the-student-that-will-replace-the-chalk/
"""
from typing import List


class Solution:
    def chalk_replacer(self, chalk: List[int], k: int) -> int:
        i, n = 0, len(chalk)
        k %= sum(chalk)
        while chalk[i] <= k:
            k -= chalk[i]
            i += 1
            if i == n:
                i = 0
        return i

File: java-algo/src/main/java/com/leetcode/Problem1894.java
package com.leetcode;

/**
 * 1894. Find the Student that Will Replace the Chalk
 * https://leetcode.com/problems/find-the-student-that-will-replace-the-chalk/
 */
public class Problem1894 {
    public int chalkReplacer(int[] chalk, int k) {
        long sum = 0;
        for (int num : chalk) {
            sum += num;
        }
        k %= sum;
        int i = 0;
        while (k >= chalk[i]) {
            k -= chalk[i];
            i = (++i) % chalk.length;
        }
        return i;
    }
}
